dcapp/app/src/main/java/com/example/michael/mqtttest/MainActivity.java

```diff
@@ -0,0 +1,287 @@
+ * button.
+ *
+ * This activity also implements the SensorEventListener to allow it to
+ * display acceleration data to the user in real time.
+ */
+public class MainActivity extends AppCompatActivity implements SensorEventListener {
+
+    private MainActivity mainActivity = this;
+
+    private Intent startBroadcastService;
+
+    private String androidId;
+    private TelephonyManager telephonyManager;
+
+    private String hostIpAddress;
+    private String connectionStatus;
+
+    private int speedSetting = 0;
+    private SeekBar speedControl;
+
+    private TextView publishRateView;
+
+    private TextView currentX;
+    private TextView currentY;
+    private TextView currentZ;
+
+    private TextView hostIpView;
+    private TextView connectionStatusView;
+
+    private SensorManager sensorManager;
+    private Sensor accelerometer;
+
+    private float lastReadX = 0;
+    private float lastReadY = 0;
+    private float lastReadZ = 0;
+
+    // Maybe we should consider making a constants class
+    private static final String IP_NEEDED_ALERT_MESSAGE = "Please enter " +
+            "the IP address of the host including the port. e.g. " +
+            "'192.168.0.164:1883'";
+
+    private static final String HOST_IP_DISPLAY_PREFIX = "Host IP: ";
+    private static final String BROADCAST_RATE_DISPLAY_PREFIX =
+            "PublishRate (ms): ";
+    private static final String CONNECTION_DISPLAY_PREFIX =
+            "Connection Status: ";
+
+    private static final String CONFIRM_BUTTON_TEXT = "Confirm";
+    private static final String CANCEL_BUTTON_TEXT = "Cancel";
+
+    private static final String DEVICE_ID_INTENT_EXTRA = "AndroidId";
+    private static final String HOST_IP_INTENT_EXTRA = "hostIp";
+    private static final String SPEED_SETTING_INTENT_EXTRA = "userSpeedInput";
+
+    private static final int X_ACCELERATION_INDEX = 0;
+    private static final int Y_ACCELERATION_INDEX = 1;
+    private static final int Z_ACCELERATION_INDEX = 2;
+
+
+    @Override
+    /**
+     * @desc - This method is run each time MainActivity is initialized. It is
+     *         responsible for initializing the BroadcastService, registering
+     *         listeners for acceleration data in this activity to present to
+     *         the user, and discovering the Android devices IMEI ID to send
+     *         to the BroadcastService to publish with all outgoing data.
+     *
+     * @param savedInstanceState - Bundle containing the state of the calling
+     *         activity. On initialization, this value is null.
+     */
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Get IMEI ID for the current device
+        final TelephonyManager telephonyManager = (TelephonyManager) getBaseContext().getSystemService(Context.TELEPHONY_SERVICE);
+        androidId = android.provider.Settings.Secure.getString(getContentResolver(), android.provider.Settings.Secure.ANDROID_ID);
```

Comment:
Future reference, long lines of code can become a pain in the ass to read (especially if they become too prolific in the code). A code remedy to this is to use hanging indentation. E.g.:

(TelephonyManager) getBaseContext().getSystemService(Context.TELEPHONY_SERVICE);

becomes
(TelephonyManager) getBaseContext()
      .getSystemService(Context.TELEPHONY_SERVICE);